### PR TITLE
autoload without hiding errors

### PR DESCRIPTION
**Pull request by Flyingmana**
### Related Pull Requests
alternative to https://github.com/OpenMage/magento-lts/pull/1817

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#1780
2. Closes OpenMage/magento-lts#1817

### Manual testing scenarios (*)

1. ...
3. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by justinbeaty** [comment]
This looks like a much better solution. Should you use the original return value?

```php
return include $found;
```

Maybe the return value not used anywhere though...

**Comment by Flyingmana** [comment]
> This looks like a much better solution. Should you use the original return value?
> 
> ```
> return include $found;
> ```
> 
> Maybe the return value not used anywhere though...

you are right, will add it again

**Review by justinbeaty — DISMISSED** [review]
Applied patch in my local dev env and all seems good!

**Comment by luigifab** [comment]
This is a simplest version of @Asfolny idea. I will test it!

**Comment by luigifab** [comment]
Some tests with blackfire.

First, with `$path = str_replace(['_', '\\', ' '], DIRECTORY_SEPARATOR, $class) . '.php';` + `return @include $path;`
image

Second, with `$path = str_replace(['_', '\\', ' '], DIRECTORY_SEPARATOR, $class) . '.php';` + `stream_resolve_include_path` of the PR
image

Third, with `$path = str_replace(['_', '\\', ' '], DIRECTORY_SEPARATOR, $class) . '.php';` + `foreach get_include_path` ([link](https://github.com/OpenMage/magento-lts/pull/1817#issuecomment-994761994))
image

The difference between `stream_resolve_include_path` (12.3 - 10.0 - 8.81 ms) and `foreach get_include_path` (10.4 - 12.1 - 10.8 ms) seems to be negligible.

File: lib/Varien/Autoload.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ public function autoload($class)
         /** @see https://stackoverflow.com/a/5504486/716029 */
         $found = stream_resolve_include_path($path);
         if ($found !== FALSE) {
```
Comment (sreichel) [inline lib/Varien/Autoload.php:68]:
```suggestion
       if ($found !== false) {
```

To avoid php-cs-fixer error.

**Review by sreichel — APPROVED** [review]
Tested.

**Review by ADDISON74 — APPROVED** [review]

**Pull request merged**